sklearn/cluster/tests/test_kernel_kmeans.py

```diff
@@ -0,0 +1,101 @@
+""" Testing for Kernel K means"""
+import sys
+
+import numpy as np
+from contextlib import contextmanager
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_array_equal
+from sklearn.utils.testing import assert_raises
+from sklearn.utils.testing import assert_raises_regex
+from sklearn.utils.testing import assert_not_equal
+
+from sklearn.cluster import KernelKMeans
+from sklearn.datasets.samples_generator import make_blobs
+from sklearn.externals.six.moves import cStringIO as StringIO
+
+centers = np.array([
+    [0.0, 5.0, 0.0, 0.0, 0.0],
+    [1.0, 1.0, 4.0, 0.0, 0.0],
+    [1.0, 0.0, 0.0, 5.0, 1.0],
+])
+n_samples = 100
+n_clusters, n_features = centers.shape
+X, true_labels = make_blobs(n_samples=n_samples, centers=centers,
+                            cluster_std=1., random_state=42)
+
+
+def test_kernel_kmeans_dtype():
+    rnd = np.random.RandomState(0)
+    X = rnd.normal(size=(40, 2))
+    X = (X * 10).astype(np.uint8)
+    clf = KernelKMeans().fit(X)
+    pred_x = clf.predict(X)
+    assert_array_equal(clf.labels_, pred_x)
+
+
+def test_predict():
+    clf = KernelKMeans(n_clusters=n_clusters, random_state=42)
+    clf.fit(X)
+
+    # sanity check: re-predict labeling for training set samples
+    pred = clf.predict(X)
+    assert_array_equal(pred, clf.labels_)
```

Comment:
You should also check that prediction is possible on a non-training dataset, and results should at least differ between datasets.